filebeat/harvester/reader/multiline.go

```diff
@@ -114,31 +114,34 @@ func (mlr *Multiline) Next() (Message, error) {
 
 func (mlr *Multiline) readFirst() (Message, error) {
 	for {
-		p, err := mlr.reader.Next()
-		if err == nil {
-			if p.Bytes == 0 {
+		message, err := mlr.reader.Next()
+		if err != nil {
+			// no lines buffered -> ignore timeout
+			// TODO: Does this only happen on backoff? Normally is EOF?
+			if err == errMultilineTimeout {
 				continue
 			}
 
-			mlr.startNewLine(p)
-			mlr.state = (*Multiline).readNext
-			return mlr.readNext()
+			// something is wrong here
+			return message, err
 		}
 
-		// no lines buffered -> ignore timeout
-		if err == errMultilineTimeout {
+		// Empty starting lines are skipped
+		if message.Bytes == 0 {
```

Comment:
`message.Bytes` normally contains number of newline characters. A message with `message.Bytes==0` is more like a zero/empty message. Question: what makes an empty line?

an empty line and a zero-message both have `len(msg.Content) == 0`. But empty line has `msg.Bytes>0`, but a zero message has `msg.Bytes==0`.